internal/extproc/chatcompletion_processor.go

```diff
@@ -352,6 +356,9 @@ func (c *chatCompletionProcessorUpstreamFilter) SetBackend(ctx context.Context,
 	c.originalRequestBodyRaw = rp.originalRequestBodyRaw
 	c.onRetry = rp.upstreamFilterCount > 1
 	c.stream = c.originalRequestBody.Stream
+	if isEndpointPicker {
+		c.logger.Debug("selected backend", slog.String("picked_endpoint", pickedEndpoint), slog.String("backendName", b.Name), slog.String("modelNameOverride", c.modelNameOverride), slog.String("originalRequestBodyRaw", string(c.originalRequestBodyRaw)))
```

Comment:
this line is crazy costly implicitly. the expression evaluation happens before the logging level check inside logger.Debug. so making string out of `string(c.originalRequestBodyRaw)` happens regardless of the log level. maybe you might want to delete this line or use `Enabled` to avoid the evaluation. Either way i don't think you would need the body here as it will be logged if the log level is set to debug somwhere in server.go
```suggestion
		if c.logger.Enabled(ctx, slog.LevelDebug) { 
				c.logger.Debug("selected backend", slog.String("picked_endpoint", pickedEndpoint), slog.String("backendName", b.Name), slog.String("modelNameOverride", c.modelNameOverride))
		}
```